pkg/util/domains/domains.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package domains
+
+import (
+	"strings"
+)
+
+// DomainFilter holds a lists of valid domain names
+type DomainFilter struct {
+	filters []string
+}
+
+// NewDomainFilter returns a new DomainFilter given a comma separated list of domains
+func NewDomainFilter(filters []string) DomainFilter {
+
```

Comment:
nitpick: it is better to `make` a new slice and copy the strings into it without retaining StringHeader